### PR TITLE
feat: spreadsheet without stream

### PR DESCRIPTION
**Pull request by IsaMic**
|Q              |A  |
|---------------|---|
|Bug fix?       |n|
|New feature?   |y|	
|BC breaks?     |n|	
|Deprecations?  |n|	
|Fixed tickets  |n|	

Need use _emsch_cache in route to generate big .xlsx => implement a version no stream

### CODE REVIEW

File: src/Common/SpreadsheetGeneratorService.php [hunk 1]
```diff
@@ -144,10 +188,39 @@ function () use ($config) {
...
+
+    /**
+     * @param Response|StreamedResponse                                                   $response
+     * @param array{writer: string, filename: string, disposition: string, sheets: array} $config
+     */
+    private function attachResponseHeader($response, array $config, string $filename, string $type): void
+    {
+        $response->headers->set('Content-Type', $type);
+        $response->headers->set('Content-Disposition', \sprintf($filename, $config[self::CONTENT_DISPOSITION], $config[self::CONTENT_FILENAME]));
+    }
```
Comment (theus77) [inline src/Common/SpreadsheetGeneratorService.php:225]:
```suggestion
    private function attachResponseHeader($response, array $config, string $extension, string $type): void
    {
        $response->headers->set('Content-Type', $type);
        $response->headers->set('Content-Disposition', \sprintf('%s;filename="%s.%s"', $config[self::CONTENT_DISPOSITION], $config[self::CONTENT_FILENAME], $extension));
    }
```

File: src/Common/SpreadsheetGeneratorService.php [hunk 2]
```diff
@@ -106,28 +135,43 @@ private function resolveOptions(array $config): array
...
         return $response;
     }
 
     /**
      * @param array{writer: string, filename: string, disposition: string, sheets: array} $config
      */
-    private function getCsvResponse(array $config): StreamedResponse
+    private function getCsvStreamedResponse(array $config): StreamedResponse
     {
+        \dump($config);
```
Comment (theus77) [inline src/Common/SpreadsheetGeneratorService.php:174]: Must be removed

File: src/Common/SpreadsheetGeneratorService.php [hunk 3]
```diff
@@ -106,28 +135,43 @@ private function resolveOptions(array $config): array
...
+
+    /**
+     * @param array{writer: string, filename: string, disposition: string, sheets: array} $config
+     */
+    private function getXlsxResponse(array $config): Response
+    {
+        $spreadsheet = $this->buildUpSheets($config);
+        $writer = new Xlsx($spreadsheet);
+
+        $writer->save($config[self::CONTENT_FILENAME]);
```
Comment (theus77) [inline src/Common/SpreadsheetGeneratorService.php:161]: Is it not better to save it in a temp file?

File: src/Contracts/SpreadsheetGeneratorServiceInterface.php [hunk 4]
```diff
@@ -19,4 +20,9 @@ interface SpreadsheetGeneratorServiceInterface
      * @param array<mixed> $config
      */
     public function generateSpreadsheet(array $config): StreamedResponse;
+
+    /**
+     * @param array<mixed> $config
+     */
+    public function generateSpreadsheetResponse(array $config): Response;
```
[the remark this answers is not in the document]
Comment (Davidmattei) [reply src/Contracts/SpreadsheetGeneratorServiceInterface.php:27]: I would not place Cacheable in the function name, the generator can give a streamed response and a response. That you want a response for cache is not to know for this generator

**Pull request merged**